upgrade/v15xto160/upgrade.go

```diff
@@ -0,0 +1,179 @@
+package v15xto160
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
+	"github.com/longhorn/longhorn-manager/types"
+	upgradeutil "github.com/longhorn/longhorn-manager/upgrade/util"
+)
+
+const (
+	upgradeLogPrefix = "upgrade from v1.5.x to v1.6.0: "
+)
+
+func UpgradeResources(namespace string, lhClient *lhclientset.Clientset, kubeClient *clientset.Clientset, resourceMaps map[string]interface{}) error {
+	if err := upgradeCSIPlugin(namespace, kubeClient); err != nil {
+		return err
+	}
+
+	if err := upgradeVolumes(namespace, lhClient, resourceMaps); err != nil {
+		return err
+	}
+
+	if err := upgradeNodes(namespace, lhClient, resourceMaps); err != nil {
+		return err
+	}
+
+	if err := upgradeEngines(namespace, lhClient, resourceMaps); err != nil {
+		return err
+	}
+
+	if err := upgradeReplicas(namespace, lhClient, resourceMaps); err != nil {
+		return err
+	}
+
+	return upgradeOrphans(namespace, lhClient, resourceMaps)
```

Comment:
Should we remove these functions since we don't upgrade any thing inside these functions?

Reply:
I assumed these were all placeholders to add steps to as various items crop up in 1.6.0 development.  It seems harmless to leave them as NOPs.

Reply:
Discussed with @PhanLe1010 and @james-munson offline and chose to remove all "placeholders" but leave a couple of comments suggesting where developers looking to add more upgrade behavior can go for examples.